packages/react/src/useActionForm.ts

```diff
@@ -0,0 +1,263 @@
+  }
+};
+
+/**
+ * The identity of a record to build a form for
+ *
+ * - Pass the ID of a record as a string to look up a record by id
+ * - Pass a {<fieldValue>: <value>} object to look up a record by some other field. __Note__: a `findBy` function must exist for the field you wish to find the record by.
+ *
+ * @example
+ * // find the user record with ID=123
+ * const { isReady, submit } = useActionForm(api.user.update, { record: "123" });
+ *
+ * @example
+ * // find a record by email. `api.user.findByEmail` must exist
+ * const { isReady, submit } = useActionForm(api.user.update, { record: { email: "user@gadget.app" } });
+ */
+export type RecordIdentifier = string | { [key: string]: any };
+
+export type UseActionFormResult<
+  GivenOptions extends OptionsType,
+  SchemaT,
+  ActionFunc extends ActionFunction<GivenOptions, any, any, SchemaT, any>,
+  FormContext = any
+> = Omit<UseFormReturn<ActionFunc["variablesType"], FormContext>, "handleSubmit"> & {
+  /**
+   * Any error that occurred during initial data fetching or action submission
+   */
+  error?: ErrorWrapper | Error | null;
+  /**
+   * Function to call to submit the form
+   */
+  submit: () => Promise<void>;
+  /**
+   * The data resulting from running the action
+   */
+  actionData?: ReturnType<typeof useAction<GivenOptions, SchemaT, ActionFunc, any>>[0]["data"];
+};
+
+/**
+ * React hook to manage state for a form that calls a Gadget action. `useActionForm` must be passed an action function from an instance of your generated API client library, like `api.user.create`, `api.blogPost.publish` or `api.someGlobalAction`. `useActionForm` returns a `Form` object from `react-hook-form` which can be used to build great form experiences.
+ *
+ * `useActionForm` manages the Gadget API calls for the form, and handling validation errors returned by the Gadget backend. If your form is for an existing record, `useActionForm` will also fetch the record from the backend and populate the form with the existing values.
+ *
+ * `useActionForm` doesn't run the action when called. To actually submit the form, call the `submit` function on the returned `Form` object.
+ *
+ * @see https://react-hook-form.com/
+ *
+ * @param action an action function from a model manager in your application's client, like `api.user.create`
+ * @param options action options, like selecting the fields in the result
+ */
+export const useActionForm = <
+  GivenOptions extends OptionsType, // currently necessary for Options to be a narrow type (e.g., `true` instead of `boolean`)
+  SchemaT,
+  ActionFunc extends ActionFunction<GivenOptions, any, any, SchemaT, any>,
+  FormContext = any
+>(
+  action: ActionFunc,
+  options?: UseFormProps<ActionFunc["variablesType"], FormContext> & {
+    /**
+     * The record identifier to run this action on, if it already exists.
+     * Should be undefined for create actions, or a record ID (or finder) for update / etc actions
+     **/
+    findBy?: RecordIdentifier;
```

Comment:
I wonder if it would be more clear with something like
```typescript
& { findBy?: RecordIdentifier; record: never } | { record?: ActionFunc["hasReturnType"] extends true ...., findBy: never }
```
not sure if that would typecheck but makes it obvious it's one or the other. Another suggestion is maybe only include these if `operatesWithRecordIentity` is true

Reply:
got rid of the record so this should be more clear now